modules/service/src/main/resources/db/migration/V0899__cfp_non_partner.sql

```diff
@@ -0,0 +1,46 @@
+-- Recreate the view to calculate non-partner flag.
+
+DROP VIEW v_cfp;
+
+CREATE VIEW v_cfp AS
+  WITH
+    cte_instrument AS (
+      SELECT
+        c.c_cfp_id,
+        array_remove(array_agg(ci.c_instrument ORDER BY ci.c_instrument), NULL) AS c_instruments,
+        array_remove(array_agg(i.c_long_name ORDER BY i.c_long_name), NULL) AS c_instrument_names
+      FROM
+        t_cfp c
+      LEFT JOIN
+        t_cfp_instrument ci ON c.c_cfp_id = ci.c_cfp_id
+      LEFT JOIN
+        t_instrument i ON i.c_tag = ci.c_instrument
+      GROUP BY
+        c.c_cfp_id
+    ),
+    cte_partner AS (
+      SELECT
+        c.c_cfp_id,
+        (COUNT(*) FILTER (WHERE p.c_deadline IS NOT NULL AND p.c_deadline > CURRENT_TIMESTAMP) > 0) AS c_is_open,
+        MAX(CASE WHEN p.c_partner = 'us' AND (c.c_type = 'regular_semester' OR c.c_type = 'directors_time') THEN 1 ELSE 0 END) AS c_non_partner,
+        MAX(CASE WHEN p.c_partner = 'us' THEN p.c_deadline ELSE NULL END) AS c_us_deadline
+      FROM
```

Comment:
The new fields are computed in the `v_cfp` view here.  Only proposals that US PIs may apply to may be used by PIs without a partner.  By default, when a CFP is created if partners are not listed then all partners are assumed.  The US deadline (which may be the default deadline if not explicitly specified) is used as the non-partner deadline.